lib/policyEvaluator/RequestContext.ts

```diff
@@ -15,9 +15,14 @@ import {
     actionMapScuba,
 } from './utils/actionMaps';
 
-const _actionNeedQuotaCheck = {
+export const actionNeedQuotaCheck = {
     objectPut: true,
+    objectPutVersion: true,
     objectPutPart: true,
+    objectRestore: true,
+    objectDelete: true,
+    multipartDelete: true,
+    objectDeleteVersion: true,
```

Comment:
not really a quota _check_ in that case : in the sense that these API must never fail due to the quota....
e.g. if quota is way higher than expected (i.e. due to overshoot or because it was lowever), the usage "after" delete may still be higher than quota, but should be accepted nonetheless...

Reply:
All delete APIs trigger quota evaluation logic - the logic, in case of quota exceeded, would first increase the inflights (scuba is not aware of the quota) and, upon quota exceeded detection, clear the last increase. In the case of deletes, we simply decrese everytime, and (in cloudserver), we explicitly ensure that any deletion always pass.

Is this comment about the variable naming?

Reply:
Yes, making sure it cannot fail may indeed be in cloudserver. So:
* variable naming here may be misleading : and could cause confusion (or bugs) when people change (cloudserver) generic code in the future...
* checking the quota "check" may be expensive and we don't need the full evaluation in case of deletes (as you mentioned), only to track the "decrease" : so it would seem more appropriate to differentiate between these case, possibly returning 3 values (do not check, check quota, decrease usage)...

Basically, by bundling all APIs together we introduce coupling between this list of API and the details of the current implementation.... which should be avoided, to facilitate future maintenance.

Reply:
Makes sense, we can split this variable into two: need quota check (increase), need quota check (decrease). Quota exceeded case will anyway require an additional call. In the case of deletes, we only issue a single call anyway.

Reply:
Done in https://github.com/scality/Arsenal/pull/2233/commits/c5ab5d034c5b988c7a5c253427c4ce6237661604